openedx/core/djangoapps/content_libraries/api/containers.py

```diff
@@ -246,14 +258,25 @@ def delete_container(
         )
     )
 
-    # TODO: trigger a LIBRARY_COLLECTION_UPDATED for each collection the container was in
+    # For each collection, trigger LIBRARY_COLLECTION_UPDATED signal and set background=True to trigger
+    # collection indexing asynchronously.
+    #
+    # To delete the container on collections
+    for collection in affected_collections:
+        LIBRARY_COLLECTION_UPDATED.send_event(
+            library_collection=LibraryCollectionData(
+                library_key=library_key,
+                collection_key=collection.key,
+                background=True,
+            )
+        )
 
 
 def restore_container(container_key: LibraryContainerLocator) -> None:
     """
     Restore the specified library container.
     """
-    container = _get_container(container_key, isDeleted=True)
+    container = get_container_from_key(container_key, isDeleted=True)
 
     authoring_api.set_draft_version(container.pk, container.versioning.latest.pk)
 
```

Comment:
We should trigger `LIBRARY_COLLECTION_UPDATED` when restoring a container that's in some collections too.

Reply:
Good catch! Fixed here: 8d92de4dc75343d8778ab9d5f19662379b70236e